examples/as-median/assembly/index.ts

```diff
@@ -0,0 +1,63 @@
+
+export function alloc(count: i32): usize {
+	/* allocate memory buffer */
+	let buf = new Array<i32>(count);
+	let buf_ptr = memory.data(32);
+
+	/*
+	 * dereference a pointer and mutate
+	 * its underlying value
+	 */
+	store<Array<i32>>(buf_ptr, buf);
+
+	return buf_ptr
+}
+
+// note that a median can have a fractional part
+export function exec(msg: usize, msg_len: usize, out_ptr: usize, out_size: usize): i32 {
+	/* error if msg size is not a multiple of 32 in bits */
+	if (msg_len % 4 != 0) { return -1 }
+	let count = msg_len / 4;
+	let val = new Array<i32>(count as i32);
+
+	/* convert length to bits from octets */
+	let len = (msg_len * 8) as i32;
```

Comment:
For the input buffer, we should probably adjust the code to expect and parse a JSON string, rather than i32 values.

Reply:
Noted!  I will have to bring in [assemblyscript-json](https://www.npmjs.com/package/assemblyscript-json) for this purpose